File: Data Structures & Algorithms/path-sum/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        def traverse(root, sum):
            if not root:
                return False
            sum += root.val

            if not root.left and not root.right:
                return sum == targetSum
            if traverse(root.left, sum):
                return True
            if traverse(root.right, sum):
                return True
            return False
        
        return traverse(root, 0)

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-3.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    private void DFS(TreeNode root, List<int> list, int level){
        if (root == null) return;

        if (level == list.Count){
            list.Add(root.val);
        }
        
        DFS(root.right, list, level+1);
        DFS(root.left, list, level+1);
    }
    
    public List<int> RightSideView(TreeNode root) {
        var res = new List<int>();
        DFS(root, res, 0);
        return res;
    }
}
